<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>AudioDevicesBase</class>
 <widget class="QMainWindow" name="AudioDevicesBase">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>679</width>
    <height>598</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Audio Devices</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QScrollArea" name="scrollArea">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="widgetResizable">
       <bool>true</bool>
      </property>
      <widget class="QWidget" name="scrollAreaWidgetContents">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>659</width>
         <height>558</height>
        </rect>
       </property>
       <layout class="QGridLayout" name="gridLayout_4">
        <item row="0" column="0">
         <layout class="QGridLayout" name="gridLayout_2">
          <item row="0" column="0">
           <widget class="QLabel" name="modeLabel">
            <property name="text">
             <string>Mode</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLabel" name="deviceLabel">
            <property name="text">
             <string>Device</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QComboBox" name="modeBox">
            <item>
             <property name="text">
              <string>Input</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Output</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QComboBox" name="deviceBox"/>
          </item>
          <item row="2" column="0" colspan="2">
           <widget class="QTabWidget" name="tabWidget">
            <property name="currentIndex">
             <number>0</number>
            </property>
            <widget class="QWidget" name="testFormatTab">
             <attribute name="title">
              <string>Test format</string>
             </attribute>
             <layout class="QGridLayout" name="gridLayout">
              <item row="0" column="1">
               <widget class="QLabel" name="actualLabel">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="frameShape">
                 <enum>QFrame::NoFrame</enum>
                </property>
                <property name="frameShadow">
                 <enum>QFrame::Plain</enum>
                </property>
                <property name="text">
                 <string>&lt;i&gt;Actual Settings&lt;/i&gt;</string>
                </property>
                <property name="textFormat">
                 <enum>Qt::RichText</enum>
                </property>
                <property name="alignment">
                 <set>Qt::AlignCenter</set>
                </property>
               </widget>
              </item>
              <item row="0" column="2">
               <widget class="QLabel" name="nearestLabel">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="frameShape">
                 <enum>QFrame::NoFrame</enum>
                </property>
                <property name="frameShadow">
                 <enum>QFrame::Plain</enum>
                </property>
                <property name="text">
                 <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'MS Shell Dlg 2'; font-size:8.25pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-style:italic;&quot;&gt;Nearest Settings&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                </property>
                <property name="textFormat">
                 <enum>Qt::RichText</enum>
                </property>
                <property name="alignment">
                 <set>Qt::AlignCenter</set>
                </property>
               </widget>
              </item>
              <item row="3" column="1">
               <widget class="QComboBox" name="sampleRateBox">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
               </widget>
              </item>
              <item row="3" column="2">
               <widget class="QLineEdit" name="nearestSampleRate">
                <property name="enabled">
                 <bool>false</bool>
                </property>
               </widget>
              </item>
              <item row="5" column="1">
               <widget class="QComboBox" name="channelsBox"/>
              </item>
              <item row="5" column="2">
               <widget class="QLineEdit" name="nearestChannel">
                <property name="enabled">
                 <bool>false</bool>
                </property>
               </widget>
              </item>
              <item row="9" column="1">
               <widget class="QComboBox" name="sampleSizesBox"/>
              </item>
              <item row="9" column="2">
               <widget class="QLineEdit" name="nearestSampleSize">
                <property name="enabled">
                 <bool>false</bool>
                </property>
               </widget>
              </item>
              <item row="14" column="1">
               <widget class="QComboBox" name="endianBox"/>
              </item>
              <item row="14" column="2">
               <widget class="QLineEdit" name="nearestEndian">
                <property name="enabled">
                 <bool>false</bool>
                </property>
               </widget>
              </item>
              <item row="15" column="1">
               <widget class="QPushButton" name="testButton">
                <property name="text">
                 <string>Test</string>
                </property>
               </widget>
              </item>
              <item row="15" column="2">
               <widget class="QLabel" name="testResult">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
              <item row="3" column="0">
               <widget class="QLabel" name="actualFreqLabel">
                <property name="text">
                 <string>Frequency (Hz)</string>
                </property>
               </widget>
              </item>
              <item row="5" column="0">
               <widget class="QLabel" name="actualChannelLabel">
                <property name="text">
                 <string>Channels</string>
                </property>
               </widget>
              </item>
              <item row="9" column="0">
               <widget class="QLabel" name="actualSampleSizeLabel">
                <property name="text">
                 <string>Sample size (bits)</string>
                </property>
               </widget>
              </item>
              <item row="14" column="0">
               <widget class="QLabel" name="actualEndianLabel">
                <property name="text">
                 <string>Endianess</string>
                </property>
               </widget>
              </item>
              <item row="16" column="0" colspan="3">
               <widget class="QLabel" name="label">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="text">
                 <string>Note: an invalid codec 'audio/test' exists in order to allow an invalid format to be constructed, and therefore to trigger a 'nearest format' calculation.</string>
                </property>
                <property name="wordWrap">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item row="2" column="0">
               <widget class="QLabel" name="actualCodecLabel">
                <property name="text">
                 <string>Codec</string>
                </property>
               </widget>
              </item>
              <item row="2" column="2">
               <widget class="QLineEdit" name="nearestCodec">
                <property name="enabled">
                 <bool>false</bool>
                </property>
               </widget>
              </item>
              <item row="2" column="1">
               <widget class="QComboBox" name="codecsBox"/>
              </item>
              <item row="6" column="0">
               <widget class="QLabel" name="actualSampleTypeLabel">
                <property name="text">
                 <string>SampleType</string>
                </property>
               </widget>
              </item>
              <item row="6" column="1">
               <widget class="QComboBox" name="sampleTypesBox"/>
              </item>
              <item row="6" column="2">
               <widget class="QLineEdit" name="nearestSampleType">
                <property name="enabled">
                 <bool>false</bool>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="tab">
             <attribute name="title">
              <string>All formats</string>
             </attribute>
             <layout class="QVBoxLayout" name="verticalLayout_2">
              <item>
               <widget class="QPushButton" name="populateTableButton">
                <property name="text">
                 <string>Populate table</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QTableWidget" name="allFormatsTable">
                <property name="editTriggers">
                 <set>QAbstractItemView::NoEditTriggers</set>
                </property>
                <property name="dragDropOverwriteMode">
                 <bool>false</bool>
                </property>
                <property name="selectionMode">
                 <enum>QAbstractItemView::NoSelection</enum>
                </property>
                <property name="selectionBehavior">
                 <enum>QAbstractItemView::SelectItems</enum>
                </property>
                <property name="textElideMode">
                 <enum>Qt::ElideNone</enum>
                </property>
                <property name="sortingEnabled">
                 <bool>false</bool>
                </property>
                <property name="wordWrap">
                 <bool>false</bool>
                </property>
                <property name="cornerButtonEnabled">
                 <bool>false</bool>
                </property>
                <attribute name="horizontalHeaderHighlightSections">
                 <bool>false</bool>
                </attribute>
                <attribute name="verticalHeaderVisible">
                 <bool>false</bool>
                </attribute>
                <attribute name="verticalHeaderHighlightSections">
                 <bool>false</bool>
                </attribute>
                <column>
                 <property name="text">
                  <string>Codec</string>
                 </property>
                 <property name="textAlignment">
                  <set>AlignHCenter|AlignVCenter|AlignCenter</set>
                 </property>
                </column>
                <column>
                 <property name="text">
                  <string>Frequency (Hz)</string>
                 </property>
                 <property name="textAlignment">
                  <set>AlignHCenter|AlignVCenter|AlignCenter</set>
                 </property>
                </column>
                <column>
                 <property name="text">
                  <string>Channels</string>
                 </property>
                 <property name="textAlignment">
                  <set>AlignHCenter|AlignVCenter|AlignCenter</set>
                 </property>
                </column>
                <column>
                 <property name="text">
                  <string>Sample type</string>
                 </property>
                 <property name="textAlignment">
                  <set>AlignHCenter|AlignVCenter|AlignCenter</set>
                 </property>
                </column>
                <column>
                 <property name="text">
                  <string>Sample size (bits)</string>
                 </property>
                 <property name="textAlignment">
                  <set>AlignHCenter|AlignVCenter|AlignCenter</set>
                 </property>
                </column>
                <column>
                 <property name="text">
                  <string>Endianness</string>
                 </property>
                 <property name="textAlignment">
                  <set>AlignHCenter|AlignVCenter|AlignCenter</set>
                 </property>
                </column>
               </widget>
              </item>
             </layout>
            </widget>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
